src/CollAction/Controllers/ProjectsController.cs

```diff
@@ -214,6 +236,25 @@ public ViewResult StartInfo()
             project.Target = editProjectViewModel.Target;
             project.End = editProjectViewModel.End;
 
+            if (editProjectViewModel.HasBannerImageUpload)
+            {
+                var manager = new ImageFileManager
+                {
+                    Context = _context,
+                    WebRoot = _hostingEnvironment.WebRootPath,
+                    WebFolder = Path.Combine("usercontent", "bannerimages")
+                };
+
+                if (project.BannerImage != null)
+                {
+                    manager.DeleteImageFile(project.BannerImage);
+                }
+
+                // Don't trust the users _formFile.FileName to be unique, so generate our own unique one "banner_<formatted-Project.Name>.<FileType>"
+                var uniqueFileName = (new Regex(@"[^A-Za-z0-9]+")).Replace("banner_" + project.Name, "_").Trim(new char[] { '_' });
```

Comment:
Maybe if we're doing this more often, add this property (NormalizedProjectName) to the model?

Also, this might be simpler: string uniqueFileName = Guid.NewGuid().ToString();

Either is fine.